src/main/java/dev/sigstore/rekor/client/RekorVerifier.java

```diff
@@ -74,12 +72,21 @@ public void verifyEntry(RekorEntry entry)
 
     // TODO: Verify more than just "ec" signed rekor entries
     // (https://github.com/sigstore/sigstore-java/issues/32)
-    var verifier = Signature.getInstance("SHA256withECDSA");
-    verifier.initVerify(rekorPublicKey);
-    verifier.update(signableJson.getBytes(StandardCharsets.UTF_8));
-    if (!verifier.verify(
-        Base64.getDecoder().decode(entry.getVerification().getSignedEntryTimestamp()))) {
-      throw new RekorVerificationException("Entry SET was not valid");
+    try {
+      var verifier = Signature.getInstance("SHA256withECDSA");
+      verifier.initVerify(rekorPublicKey);
+      verifier.update(signableJson.getBytes(StandardCharsets.UTF_8));
+      if (!verifier.verify(
+          Base64.getDecoder().decode(entry.getVerification().getSignedEntryTimestamp()))) {
+        throw new RekorVerificationException("Entry SET was not valid");
+      }
+    } catch (InvalidKeyException ike) {
+      throw new RekorVerificationException("Public Key was invalid", ike);
+    } catch (SignatureException se) {
+      throw new RekorVerificationException("Signature was invalid", se);
+    } catch (NoSuchAlgorithmException nsae) {
+      // we should never be here, the verifier algorithm is expected to exist
+      throw new RuntimeException(nsae);
```

Comment:
the comment could go as a message into the exception.  You could also use AssertionError to make it clearer this shouldn't happen.

Reply:
Word

Reply:
done